source/common/network/utility.cc

```diff
@@ -52,6 +52,18 @@ Address::InstanceConstSharedPtr Utility::resolveUrl(const std::string& url) {
   }
 }
 
+Socket::Type Utility::socketTypeFromUrl(const std::string& url) {
```

Comment:
Perhaps add a couple of unit tests for this function?

Reply:
Sure, added tests for the function